helix-core/src/main/java/org/apache/helix/task/JobRebalancer.java

```diff
@@ -424,9 +425,15 @@ private ResourceAssignment computeResourceMapping(String jobResource,
             .contains(instance)) {
           continue;
         }
+        // 1. throttled by job configuration
         // Contains the set of task partitions currently assigned to the instance.
         Set<Integer> pSet = entry.getValue();
-        int numToAssign = jobCfg.getNumConcurrentTasksPerInstance() - pSet.size();
+        int jobCfgLimitation = jobCfg.getNumConcurrentTasksPerInstance() - pSet.size();
```

Comment:
I get your point.
But it is not quite necessary at this moment.
I prefer to split this huge method when there is any re-used logic.